pay-api/src/pay_api/services/payment_account.py

```diff
@@ -556,6 +561,52 @@ def find_by_id(cls, account_id: int):
         account._dao = PaymentAccountModel.find_by_id(account_id)  # pylint: disable=protected-access
         return account
 
+    @classmethod
+    def get_eft_credit_balance(cls, account_id: int) -> Decimal:
+        """Calculate pay account eft balance by account id."""
+        result = db.session.query(func.sum(EFTCreditModel.remaining_amount).label('credit_balance')) \
+            .filter(EFTCreditModel.payment_account_id == account_id) \
+            .group_by(EFTCreditModel.payment_account_id) \
+            .one_or_none()
+
+        return Decimal(result.credit_balance) if result else 0
+
+    @classmethod
+    def deduct_eft_credit(cls, account_id: int, invoice: Invoice) -> Decimal:
+        """Deduct EFT credit and update remaining credit records."""
+        # Get all eft credits
+        result: List[EFTCreditModel] = db.session.query(EFTCreditModel) \
```

Comment:
suggestion rename `result` -> `eft_credits`

I don't think you'd need the comment on 577 then

Reply:
Done